packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,265 @@
+    _sensitiveContentService.setContentSensitivity(desiredSensitivityLevel);
+
+    // Update current content sensitivity level.
+    calculatedContentSensitivityLevel = desiredSensitivityLevel;
+  }
+
+  /// Unregisters a [SensitiveContent] widget from the [ContentSensitivitySetting] tracking
+  /// the content sensitivity level of the widget tree.
+  static void unregister(ContentSensitivity widgetSensitivityLevel) {
+    instance._unregister(widgetSensitivityLevel);
+  }
+
+  void _unregister(ContentSensitivity widgetSensitivityLevel) {
+    assert(
+      _contentSenstivityIsSupported != null,
+      'SensitiveContentHost.register must be called before SensitiveContentHost.unregister',
+    );
+
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    _contentSensitivitySetting.removeWidgetWithContentSensitivity(widgetSensitivityLevel);
+
+    if (!_contentSensitivitySetting.hasWidgets) {
+      // Restore default content sensitivity setting if there are no more SensitiveContent
+      // widgets in the tree. If the call to set content sensitivity on the platform side fails,
+      // then we do not update the current content sensitivity level. The null check is safe
+      // because _fallbackContentSensitivitySetting cannot be null if `register` has been
+      // called at least once, and it must be called before `unregister` is called.
+      _sensitiveContentService.setContentSensitivity(_fallbackContentSensitivitySetting!);
+
+      // Update current content sensitivity level.
+      calculatedContentSensitivityLevel = _fallbackContentSensitivitySetting;
+      return;
+    }
+
+    // Determine if another sensitivity level needs to be restored. The null check should be
+    // safe because contentSensitivityBasedOnWidgetCounts should always be non-null as long
+    // as there are still SensitiveContent widgets in the tree.
+    final ContentSensitivity contentSensitivityToRestore =
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts!;
+    if (contentSensitivityToRestore != calculatedContentSensitivityLevel) {
+      // Set content sensitivity level as contentSensitivityToRestore. If the call to set content
+      // sensitivity on the platform side fails, then we do not update the current content
+      // sensitivity level.
+      _sensitiveContentService.setContentSensitivity(contentSensitivityToRestore);
+
+      // Update current content sensitivity level.
+      calculatedContentSensitivityLevel = contentSensitivityToRestore;
+    }
+  }
+}
+
+/// Widget to set the [ContentSensitivity] level of content in the widget
+/// tree.
+///
+/// {@macro flutter.services.ContentSensitivity}
+///
+/// See also:
+///
+///  * [ContentSensitivity], which are the different content sensitivity levels that a
+///    [SensitiveContent] widget can set.
```

Comment:
Add something about what the behavior is on platforms that dont support sensitive content and how/where to tell if sensitive content is supported.